fairlearn/postprocessing/plotting.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Utilities for plotting curves."""
+
+from ._constants import _MATPLOTLIB_IMPORT_ERROR_MESSAGE, DEMOGRAPHIC_PARITY, EQUALIZED_ODDS
+from ._threshold_optimizer import ThresholdOptimizer
+
+try:
+    import matplotlib.pyplot as plt
+    import matplotlib.cm as cm
+    import matplotlib.colors
+
+    tab10_norm = matplotlib.colors.Normalize(vmin=0, vmax=7)
+    tab10_scalarMap = cm.ScalarMappable(norm=tab10_norm, cmap='Dark2')
+    debug_colors = [tab10_scalarMap.to_rgba(x) for x in range(10)]
+    debug_ncolors = len(debug_colors)
+    debug_colormap = {}
+
+    highlight_color = [0.95, 0.90, 0.40]
+except ImportError:
+    raise RuntimeError(_MATPLOTLIB_IMPORT_ERROR_MESSAGE)
+
+
+def _get_debug_color(key):
+    if key not in debug_colormap:
+        color = debug_colors[len(debug_colormap) % debug_ncolors]
+        debug_colormap[key] = color
+    return debug_colormap[key]
+
+
+def _plot_solution(x_best, y_best, solution_label, xlabel, ylabel):
+    """Plot the given solution with appropriate labels."""
+    if y_best is None:
+        plt.axvline(x=x_best, label=solution_label, ls='--')
+    else:
+        plt.plot(x_best, y_best, 'm*', ms=10, label=solution_label)
+    plt.legend()
+    plt.xlabel(xlabel)
+    plt.ylabel(ylabel)
+
+
+def _plot_overlap(x_grid, y_min):
+    """Plot the overlap region."""
+    line, = plt.plot(x_grid, y_min, color=highlight_color, lw=8, label='overlap')
+    line.zorder -= 1
+
+
+def _plot_curve(sensitive_feature, x_col, y_col, points):
+    """Plot the given curve with labels."""
+    color = _get_debug_color(sensitive_feature)
+    plt.plot(points[x_col], points[y_col], c=color, ls='-', lw=2.0,
+             label='sensitive feature value ' + str(sensitive_feature))
+
+
+def _raise_if_not_threshold_optimizer(obj):
+    if not isinstance(obj, ThresholdOptimizer):
+        raise ValueError("Argument {} needs to be of type {}."
+                         .format(obj.__name__, ThresholdOptimizer.__name__))
+
+
+def plot_selection_error_curve(threshold_optimizer, show_plot=True):
```

Comment:
I'm not an expert in matplotlib, but isn't there some sort of 'plotting' object which could be passed in, rather than just using the default? I'm thinking of cases where a user might want to make these a subplot in one of their larger plots. We would have the `plt` object you use as the default.

Reply:
you can pass an `axis` around, and if it's `None`, use the default device. Plotting on an `axis` makes it easier for the user to plot on a part of a larger plot as well.

Reply:
Sounds good. Adjusting...